modules/frontend/combiner/response_metrics.go

```diff
@@ -0,0 +1,95 @@
+package combiner
+
+import (
+	"github.com/grafana/tempo/pkg/tempopb"
+)
+
+// These structs combine response metrics in a single place
+
+type SearchMetricsCombiner struct {
+	Metrics *tempopb.SearchMetrics
+}
+
+func NewSearchMetricsCombiner() *SearchMetricsCombiner {
+	return &SearchMetricsCombiner{
+		Metrics: &tempopb.SearchMetrics{},
+	}
+}
+
+func (mc *SearchMetricsCombiner) Combine(newMetrics *tempopb.SearchMetrics, resp PipelineResponse) {
+	if newMetrics != nil && !IsCacheHit(resp.HTTPResponse()) {
+		mc.Metrics.InspectedTraces += newMetrics.InspectedTraces
+		mc.Metrics.InspectedBytes += newMetrics.InspectedBytes
+		mc.Metrics.CompletedJobs++
```

Comment:
i think we should always increment this. otherwise these values might not line up in the frontend.

![image](https://github.com/user-attachments/assets/e9ae74c3-d09b-4927-90ee-b451007ba33c)